src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -0,0 +1,589 @@
+            {
+                if (_output != null)
+                {
+                    _output.WaitUtilEOF();
+                }
+                if (_error != null)
+                {
+                    _error.WaitUtilEOF();
+                }
+            }
+
+            return exited;
+        }
+
+        /// <summary>Gets the main module for the associated process.</summary>
+        public ProcessModule MainModule
+        {
+            get { throw new PlatformNotSupportedException(); }
+        }
+
+        /// <summary>Checks whether the process has exited and updates state accordingly.</summary>
+        private void UpdateHasExited()
+        {
+            int? exitCode;
+            _exited = GetWaitState().GetExited(out exitCode);
+            if (_exited && exitCode != null)
+            {
+                _exitCode = exitCode.Value;
+            }
+        }
+
+        /// <summary>Gets the time that the associated process exited.</summary>
+        private DateTime ExitTimeCore
+        {
+            get { return GetWaitState().ExitTime; }
+        }
+
+        /// <summary>Gets the amount of time the process has spent running code inside the operating system core.</summary>
+        public TimeSpan PrivilegedProcessorTime
+        {
+            get
+            {
+                return TicksToTimeSpan(GetStat().stime);
+            }
+        }
+
+        /// <summary>Gets the time the associated process was started.</summary>
+        public DateTime StartTime
+        {
+            get
+            {
+                return BootTimeToDateTime(GetStat().starttime);
+            }
+        }
+
+        /// <summary>
+        /// Gets the amount of time the associated process has spent utilizing the CPU.
+        /// It is the sum of the <see cref='System.Diagnostics.Process.UserProcessorTime'/> and
+        /// <see cref='System.Diagnostics.Process.PrivilegedProcessorTime'/>.
+        /// </summary>
+        public TimeSpan TotalProcessorTime
+        {
+            get
+            {
+                Interop.procfs.ParsedStat stat = GetStat();
+                return TicksToTimeSpan(stat.utime + stat.stime);
+            }
+        }
+
+        /// <summary>
+        /// Gets the amount of time the associated process has spent running code
+        /// inside the application portion of the process (not the operating system core).
+        /// </summary>
+        public TimeSpan UserProcessorTime
+        {
+            get
+            {
+                return TicksToTimeSpan(GetStat().utime);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the associated process priority
+        /// should be temporarily boosted by the operating system when the main window
+        /// has focus.
+        /// </summary>
+        private bool PriorityBoostEnabledCore
+        {
+            get { return false; } //Nop
+            set { } // Nop
```

Comment:
I think the getter should return false.  I was 50/50 on the setter; since you seem to prefer the exception, I've changed it to throw PlatformNotSupportedException.